rust/agents/validator/src/validator.rs

```diff
@@ -112,22 +120,26 @@ impl BaseAgent for Validator {
 
         let reorg_period = NonZeroU64::new(self.reorg_period);
 
-        // Ensure that the mailbox has count > 0 before we begin indexing
+        // Ensure that the merkle tree hook has count > 0 before we begin indexing
         // messages or submitting checkpoints.
-        while self
-            .mailbox
-            .count(reorg_period)
-            .await
-            .expect("Failed to get count of mailbox")
-            == 0
-        {
-            info!("Waiting for first message to mailbox");
-            sleep(self.interval).await;
-        }
-
-        tasks.push(self.run_message_sync().await);
-        for checkpoint_sync_task in self.run_checkpoint_submitters().await {
-            tasks.push(checkpoint_sync_task);
+        loop {
+            match self.merkle_tree_hook.count(reorg_period).await {
+                Ok(0) => {
+                    info!("Waiting for first message in merkle tree hook");
+                    sleep(self.interval).await;
+                }
+                Ok(_) => {
+                    tasks.push(self.run_message_sync().await);
+                    for checkpoint_sync_task in self.run_checkpoint_submitters().await {
+                        tasks.push(checkpoint_sync_task);
+                    }
+                    break;
+                }
+                _ => {
+                    // Future that immediately resolves
+                    return tokio::spawn(ready(Ok(()))).instrument(info_span!("Validator"));
```

Comment:
why this? feels weird, unsure if this is a common practice though

Reply:
hmm, this is a leftover from before the merkle tree artifact was added. If querying `count` fails on the merkle tree trait object fails, the validator assumes it can gracefully terminate because it's not required to sign anything